core/src/toga/app.py

```diff
@@ -82,16 +78,13 @@ def __call__(self, app: App, /, **kwargs: Any) -> object:
 
 
 class WidgetRegistry:
-    # WidgetRegistry is implemented as a wrapper around a WeakValueDictionary, because
-    # it provides a mapping from ID to widget. The mapping is weak so the registry
-    # doesn't retain a strong reference to the widget, preventing memory cleanup.
-    #
+    # WidgetRegistry is implemented as a wrapper around a dict.
     # The lookup methods (__getitem__(), __iter__(), __len()__, keys(), items(), and
-    # values()) are all proxied to underlying data store. Private methods exist for
+    # values()) are all proxied to the underlying data store. Mutation methods exist for
     # internal use, but those methods shouldn't be used by end-users.
 
     def __init__(self, *args: Any, **kwargs: Any):
-        self._registry = WeakValueDictionary(*args, **kwargs)
+        self._registry = dict(*args, **kwargs)
```

Comment:
We need to be careful that this doesn't introduce leaks elsewhere. This was a WeakValueDict historically to ensure that there wasn't a reference loop between the ObjC and Python objects; that *might* not be an issue any more given the work done on the window/widget lifecycle, plus the in-flight Rubicon reference handling changes - but we should confirm this is true.

Reply:
The registry was made a WeakValueDict in #2066 because at that time it stored references to all widgets which *existed*. But then #2320 and #2517 tightened that up to include only widgets which are attached to a window which has not yet been closed. 

That means that if a widget is in the registry, it must also be reachable via a chain of strong references from `App.windows`. So I don't believe the weakref functionality is being used anymore – widgets are being removed from the registry before their weakrefs expire.

Reply:
I suspect you're correct here - and the memory leak tests in the testbed *should* confirm this.